exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/MapUtility.java

```diff
@@ -228,11 +228,301 @@ public static void writeToMapFromReader(FieldReader fieldReader, BaseWriter.MapW
+      MinorType valueMinorType = valueMajorType.getMinorType();
+      boolean repeated = false;
+
+      if (valueMajorType.getMode() == TypeProtos.DataMode.REPEATED) {
+        repeated = true;
+      }
+
+      switch (valueMinorType) {
+        case TINYINT:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).tinyInt());
+          } else {
+            fieldReader.copyAsValue(mapWriter.tinyInt(fieldName));
+          }
+          break;
+        case SMALLINT:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).smallInt());
+          } else {
+            fieldReader.copyAsValue(mapWriter.smallInt(fieldName));
+          }
+          break;
+        case BIGINT:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).bigInt());
+          } else {
+            fieldReader.copyAsValue(mapWriter.bigInt(fieldName));
+          }
+          break;
+        case INT:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).integer());
+          } else {
+            fieldReader.copyAsValue(mapWriter.integer(fieldName));
+          }
+          break;
+        case UINT1:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).uInt1());
+          } else {
+            fieldReader.copyAsValue(mapWriter.uInt1(fieldName));
+          }
+          break;
+        case UINT2:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).uInt2());
+          } else {
+            fieldReader.copyAsValue(mapWriter.uInt2(fieldName));
+          }
+          break;
+        case UINT4:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).uInt4());
+          } else {
+            fieldReader.copyAsValue(mapWriter.uInt4(fieldName));
+          }
+          break;
+        case UINT8:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).uInt8());
+          } else {
+            fieldReader.copyAsValue(mapWriter.uInt8(fieldName));
+          }
+          break;
+        case DECIMAL9:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).decimal9());
+          } else {
+            fieldReader.copyAsValue(mapWriter.decimal9(fieldName));
+          }
+          break;
+        case DECIMAL18:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).decimal18());
+          } else {
+            fieldReader.copyAsValue(mapWriter.decimal18(fieldName));
+          }
+          break;
+        case DECIMAL28SPARSE:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).decimal28Sparse());
+          } else {
+            fieldReader.copyAsValue(mapWriter.decimal28Sparse(fieldName));
+          }
+          break;
+        case DECIMAL38SPARSE:
```

Comment:
Please add `VARDECIMAL` type to this switch case. Also please note that for this type should be specified scale and precision when list writer is created.

Reply:
Done